tests/fake_datacite_client.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 Northwestern University.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite DOI Client."""
+
+from unittest.mock import Mock
+
+from flask_babelex import lazy_gettext as _
+from idutils import normalize_doi
+
+from invenio_rdm_records.services.pids import providers
+
+
+class FakeDataCiteRESTClient:
+    """DataCite REST API client wrapper."""
+
+    def __init__(self, username, password, prefix, test_mode=False, url=None,
+                 timeout=None):
+        """Initialize the REST client wrapper.
+
+        :param username: DataCite username.
+        :param password: DataCite password.
+        :param prefix: DOI prefix (or CFG_DATACITE_DOI_PREFIX).
+        :param test_mode: use test URL when True
+        :param url: DataCite API base URL.
+        :param timeout: Connect and read timeout in seconds. Specify a tuple
+            (connect, read) to specify each timeout individually.
+        """
+        self.username = str(username)
+        self.password = str(password)
+        self.prefix = str(prefix)
+
+        if test_mode:
+            self.api_url = "https://api.test.datacite.org/"
+        else:
+            self.api_url = url or "https://api.datacite.org/"
+
+        if not self.api_url.endswith('/'):
+            self.api_url += '/'
+
+        self.timeout = timeout
+
+    def public_doi(self, metadata, url, doi=None):
+        """Create a public doi ... not.
```

Comment:
![image](https://user-images.githubusercontent.com/6437519/150525431-c82121b4-626a-49ed-8067-0f3079cec73c.png)

Reply:
:smile: I couldn't put the exclamation point because of pycodestyle but I am glad someone got it.